### PR TITLE
#160577440 implement login with social accounts

### PR DESCRIPTION
**Pull request by caveinn**
## What does this PR do?
- Have the implementation for social authentication

## Description of Task to be completed?
- Allow authentication with facebook
- Allow authentication with google

## How should this be manually tested?
-  visit the link for the pr pipeline app [here](https://ah-shakas-frontend-stagi-pr-17.herokuapp.com/login) 

## Any background context you want to provide?
N/A
## What are the relevant pivotal tracker stories?
[#160577440](https://www.pivotaltracker.com/n/projects/2198533/stories/160577440)
## Packages installed
 - react-social-login
 - redux-promise-middleware

## Deployment note
N/A

## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------

## Todos
- [x] Raise PR
- [x ] Test

## Screenshots (if appropriate)
image

### CODE REVIEW

**Comment by Arthur236** [comment]
@caveinn kindly work on fixing the linting issues

File: src/__test__/components/SocialAuthComponent.test.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import SocialAuth, { buttonsocial } from '../../components/Auth/socialauth.component';
+
+describe('test the social auth container', () => {
+  it('test google button renders', () => {
+    const wrapper = shallow(<SocialAuth />);
+    expect(wrapper.find('n').length).toEqual(2);
+  });
+  it('test button is rendered', () => {
```
Comment (Arthur236) [inline src/__test__/components/SocialAuthComponent.test.js]: Have a blank line to separate each of your `it` blocks

File: src/components/Auth/socialauth.component.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
```
Comment (Arthur236) [inline src/components/Auth/socialauth.component.jsx]: Rename this file as per the conventions we agreed to

File: src/constants/SocialAuthTypes.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const SOCIALLOGIN = 'SOCIALLOGIN';
```
Comment (Arthur236) [inline src/constants/SocialAuthTypes.js]: Have these constants in the right file as we had discussed

File: src/containers/Auth/socialauth.container.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { connect } from 'react-redux';
```
Comment (Arthur236) [inline src/containers/Auth/socialauth.container.js]: Rename this file as per the conventions we agreed to

File: src/reducers/SocialAuthReducer.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { SOCIALLOGIN } from '../constants/SocialAuthTypes';
```
Comment (Arthur236) [inline src/reducers/SocialAuthReducer.jsx]: Rename this file as per the conventions we agreed to

File: src/reducers/index.js [hunk 6]
```diff
@@ -1,8 +1,9 @@
 import { combineReducers } from 'redux';
 import signUp from './signup.reducer';
+import SocialAuthReducer from './SocialAuthReducer';
 
 const rootReducer = combineReducers({
-  signUp,
+  signUp, social: SocialAuthReducer,
```
Comment (Arthur236) [inline src/reducers/index.js]: Place each reducer on its own line

**Comment by Arthur236** [comment]
screenshot 2018-11-12 at 08 39 13

Make the margins equal

**Comment by caveinn** [comment]
> screenshot 2018-11-12 at 08 39 13
> 
> Make the margins equal

The reason I have not fixed this yet is because this is a dummy login page. Am hoping to style it to fit into the actual login page once it is merged.

**Review by Arthur236 — APPROVED** [review]
Looks good 👍

**Pull request merged**